infrastructure/doc_processing.py

```diff
@@ -98,6 +114,18 @@ def generate_processing_doc():
                     option += 'Min: ' + str(param.minimum()) + ', '
                 if param.maximum():
                     option += 'Max: ' + str(param.maximum())
+            elif isinstance(param, QgsProcessingParameterVectorLayer):
+                name_type = 'Type: '
+                count = 0
+                for item in param.dataTypes():
```

Comment:
* Plus pythonique si tu utilises des `enumerate` au lieu de déclarer une variable `count=0`.
* Et encore plus pythonique : `', '.join([dict_type[item] for item in param.dataTypes()])`
ou en 2 lignes : 
```python
name_types = [dict_type[item] for item in param.dataTypes()]
option += ', '.join(name_types)

Reply:
`name_types += [dict_type[item] for item in param.dataTypes()]`